### PR TITLE
[HIP] parallel scan with more than 256 threads

### PR DESCRIPTION
**Pull request by Rombur**
This PR allows to use work groups of more than 256 for parallel scan (this should also fix the same problem in #2948 but I haven't tried yet). However with this PR, the test `hip.atomic_operations_int` fails with 
`C++ exception with description "Missing metadata for __global__ function: _ZN6Kokkos12Experimental4ImplL32hip_parallel_launch_local_memoryINS_4Impl11ParallelForIN20TestAtomicOperations11InitFunctorIiNS0_3HIPEEENS_11RangePolicyIJS7_EEES7_EEEEvT_" thrown in the test body.`
Before I spend more time to debug the problem, I was wondering: @arghdos is this a good idea to allow work groups of 1024 or should we keep the size as is?

### CODE REVIEW

File: core/src/HIP/Kokkos_HIP_Parallel_Range.hpp [hunk 1]
```diff
@@ -550,14 +550,9 @@ class ParallelScanHIPBase {
...
     if (nwork) {
-      // FIXME_HIP we cannot choose it larger for large work sizes to work
-      // correctly, the unit tests fail with wrong results
-      const int gridMaxComputeCapability_2x = 0x01fff;
-
-      // FIXME_HIP block sizes greater than 256 don't work correctly,
-      // the unit tests fail with wrong results
-      const int block_size =
-          std::min(static_cast<int>(local_block_size(m_functor)), 256);
+      const int gridMaxComputeCapability_2x = 0x0ffff;
```
Comment (masterleinad) [inline core/src/HIP/Kokkos_HIP_Parallel_Range.hpp:553]: If I revert this change, all the tests are passing for me.

**Comment by arghdos** [comment]
Sorry, I missed this one.
I suspect this is related to a known bug on our end, let's discuss in our
meeting this morning.

**Comment by dalg24** [comment]
Retest this please
